test/e2e/specs/authentication/authentication__google.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @group authentication
+ * @group calypso-release
+ *
+ * Google blocks Chrome-based browsers that are controlled via automation
+ * from performing login to their services.
+ * The workaround is to use a non-Google browser, such as Firefox.
+ * See: https://stackoverflow.com/questions/66209119/automation-google-login-with-python-and-selenium-shows-this-browser-or-app-may
+ * @browser firefox
+ */
+
+import {
+	DataHelper,
+	LoginPage,
+	SecretsManager,
+	GoogleLoginPage,
+	TOTPClient,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Authentication: Google' ), function () {
+	const credentials = SecretsManager.secrets.testAccounts.googleLoginUser;
+	let page: Page;
+	let googlePopupPage: Page;
+	let loginPage: LoginPage;
+	let googleLoginPage: GoogleLoginPage;
+	let challenge = false;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+	} );
+
+	it( 'Navigate to Login page', async function () {
+		loginPage = new LoginPage( page );
+		await loginPage.visit();
+	} );
+
+	it( 'Click on Google button', async function () {
+		googlePopupPage = await loginPage.clickLoginWithGoogle();
+	} );
+
+	it( 'Enter Google username', async function () {
+		googleLoginPage = new GoogleLoginPage( googlePopupPage );
+		await googleLoginPage.enterUsername( credentials.username );
+		await googleLoginPage.clickButton( 'Next' );
+	} );
+
+	it( 'Enter Google password', async function () {
+		await googleLoginPage.enterPassword( credentials.password );
+		await googleLoginPage.clickButton( 'Next' );
+
+		try {
+			await googlePopupPage.waitForEvent( 'close', { timeout: 5 * 1000 } );
```

Comment:
I'm sure there's an even better approach that does not waste 5 seconds in case the 2FA challenge isn't shown, but I feel that a simple `waitForEvent` is legible, maintainable and solves the problem reasonably well.